graspologic/plot/plot.py

```diff
@@ -673,6 +677,253 @@ def pairplot(
+                [mean[j], mean[k]], v[0], v[1], 180.0 + angle, color=colors[i]
+            )
+        ell.set_clip_box(ax.bbox)
+        ell.set_alpha(alpha)
+        ax.add_artist(ell)
+        ax.set_xticks([]), ax.set_yticks([])
+        ax.set_ylabel(k), ax.set_xlabel(k)
+
+
+def pairplot_with_gmm(
+    X,
+    gmm,
+    labels=None,
+    title=None,
+    context="talk",
+    font_scale=1,
+    alpha=0.7,
+    palette="Set1",
+):
+    r"""
+    Plot pairwise relationships in a dataset.
+    By default, this function will create a grid of Axes such that each
+    dimension in data will by shared in the y-axis across a single row and
+    in the x-axis across a single column.
+
+    Parameters
+    ----------
+    X : np.ndarray (2D)
+        input graph
+    gmm: Gaussian Mixture Model object
+    labels : 1d np.ndarray or list, same length as dimensions of X
+        Labels for different categories of graph nodes
+    title : string, default : 'Degree plot'
+        Plot title
+    context :  None, or one of {talk (default), paper, notebook, poster}
+        Seaborn plotting context
+    font_scale : float, optional, default: 1
+        Separate scaling factor to independently scale the size of the font
+        elements.
+    alpha : float, optional, default: 0.7
+        Opacity value of plotter markers between 0 and 1
+    palette : str, dict, optional, default: 'Set1'
+        Set of colors for mapping the ``hue`` variable. If a dict, keys should
+        be values in the hue variable.
+        For acceptable string arguments, see the palette options at
+        :doc:`Choosing Colormaps in Matplotlib <tutorials/colors/colormaps>`.
+    """
+    sns.despine(left=True)
+    colors = sns.color_palette(palette)
+    # Handle X
+    if not isinstance(X, (list, np.ndarray)):
+        msg = "X must be array-like, not {}.".format(type(X))
+        raise TypeError(msg)
+
+    # Handle labels
+    if labels is not None:
+        if not isinstance(labels, (list, np.ndarray)):
+            msg = "Y must be array-like or list, not {}.".format(type(labels))
+            raise TypeError(msg)
+        elif X.shape[0] != len(labels):
+            msg = "Expected length {}, but got length {} instead for Y.".format(
+                X.shape[0], len(labels)
+            )
+            raise ValueError(msg)
+    # Handle gmm
+    if gmm is None:
+        msg = "You must input a sklearn.mixture.GaussianMixture"
+        raise TypeError(msg)
+    # get relevant features from the gmm model
+    Y_, means, covariances = gmm.predict(X), gmm.means_, gmm.covariances_
+    # plot only a scatter plot with only two feature columns
+    lab_dict, ind_to_lab = {}, {}
+    if labels:
+        set_labs = sorted(set(labels))
+        counter = 0
+        for lab in set_labs:
+            lab_dict[lab] = counter
+            ind_to_lab[counter] = lab
+            counter += 1
+        Y_ = np.asarray([lab_dict[labels[i]] for i in range(len(labels))])
```

Comment:
what does this do? just give indices in reversed order? I feel like there has to be a clearer and simpler way to write this code

Reply:
this turned the labels into numbers. it was old implementation that i fixed in this next push.